### PR TITLE
make ocr key bundle ID backwards compatible

### PR DESCRIPTION
**Pull request by RyanRHall**

### CODE REVIEW

File: core/services/keystore/keys/ocrkey/key_bundle.go [hunk 1]
```diff
@@ -101,6 +94,7 @@ func NewKeyBundleFrom(onChainSigning io.Reader, offChainSigning io.Reader, offCh
 	if err != nil {
 		return nil, err
 	}
+	fmt.Println("marshalledPrivK 1", string(marshalledPrivK))
```
Comment (samsondav) [inline core/services/keystore/keys/ocrkey/key_bundle.go:97]: Is it meant to do this?

**Review by samsondav — COMMENTED** [review]
Which is the part that actually preserves the id?

Also, do we need to worry about this for other types of keys?

**Review by samsondav — CHANGES_REQUESTED** [review]
Are all other IDs preserved on migration?

**Comment by RyanRHall** [comment]
@samsondav 
> Which is the part that actually preserves the id?
the OCR bundle's ID is just `sha256.Sum256(key.Raw())`. In this PR, I'm changing how the raw key gets encoded / decoded to match what we did before. Since the encoding is the same, the ID is the same.

> Are all other IDs preserved on migration?
Effectively, yes. No other keys actually have IDs in the same way a OCR key bundle does. Eth keys are identified by the address. P2P keys are identified by the peerID. VRF keys are identified by the address. Those all stay the same. They keys _will lose_ their row IDs, but we never used those so it's not an issue.

**Review by samsondav — APPROVED** [review]

**Pull request merged**